apps/agreement/contracts/Agreement.sol

```diff
@@ -1212,6 +1242,8 @@ contract Agreement is IArbitrable, ILockManager, IAgreement, IACLOracle, AragonA
     * @param _amount Amount of `_arbitrationFeeToken` tokens to be approved
     */
     function _approveFor(ERC20 _token, address _to, uint256 _amount) internal {
+        // Note: we may want to expose a permissionless escape hatch for resetting this app's approvals in case weird things happen on the other side
+        // E.g. arbitrator or fees cashier
         require(_token.safeApprove(_to, _amount), ERROR_TOKEN_APPROVAL_FAILED);
```

Comment:
I'm a bit hesitant about this, after ee1524c8b194cbd5a2bbdb2ad11a33f0b453e898 it seems pretty clear which are the possible approvals we have from this contract, and now that it is generalized we shouldn't have a problem with approvals. I think I'd go like this, otherwise we could open a door we don't know about yet :/

Reply:
Sounds good—I was thinking that if we could avoid the approval reset we'd save a bit of gas but it's unlikely to be a big deal.